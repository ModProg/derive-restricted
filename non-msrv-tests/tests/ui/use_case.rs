use derive_where::DeriveWhere;

#[derive(DeriveWhere)]
#[derive_where(Clone)]
struct StructNone(u8);

#[derive(DeriveWhere)]
#[derive_where(Clone)]
enum EnumNone {
	A(u8),
}

fn main() {}
