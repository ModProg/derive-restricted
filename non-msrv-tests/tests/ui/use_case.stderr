error: derive-where doesn't support items without generics, `skip` attributes or `enum`s implementing `Default`, as this can already be handled by standard `#[derive(..)]`
 --> tests/ui/use_case.rs:4:1
  |
4 | / #[derive_where(Clone)]
5 | | struct StructNone(u8);
  | |______________________^

error: derive-where doesn't support items without generics, `skip` attributes or `enum`s implementing `Default`, as this can already be handled by standard `#[derive(..)]`
  --> tests/ui/use_case.rs:8:1
   |
8  | / #[derive_where(Clone)]
9  | | enum EnumNone {
10 | |     A(u8),
11 | | }
   | |_^
